SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java

```diff
@@ -1,261 +1,601 @@
+//				.values();
+//		List<Uni<Void>> unis = Lists.newArrayList();
+//		logger.info("checking subscriptions");
+//		for (SubscriptionRequest potentialSub : potentialSubs) {
+//			switch (message.getRequestType()) {
+//				case AppConstants.UPDATE_REQUEST:
+//					if (shouldFire(message.getPayload().keySet(), potentialSub)) {
+//						unis.add(localEntityService.getEntityById(message.getTenant(), message.getId(), true).onItem()
+//								.transformToUni(entity -> {
+//									return sendNotification(potentialSub, entity, message.getRequestType());
+//								}));
+//					}
+//					break;
+//				case AppConstants.CREATE_REQUEST:
+//					unis.add(localEntityService.getEntityById(message.getTenant(), message.getId(), true).onItem()
+//							.transformToUni(entity -> {
+//								return sendNotification(potentialSub, entity, message.getRequestType());
+//							}));
+//					break;
+//				case AppConstants.DELETE_REQUEST:
+//					unis.add(sendNotification(potentialSub, message.getPayload(), message.getRequestType()));
+//				case AppConstants.DELETE_ATTRIBUTE_REQUEST:
+//					if (shouldFire(Sets.newHashSet(((DeleteAttributeRequest) message).getAttribName()), potentialSub)) {
+//						unis.add(sendNotification(potentialSub, message.getPayload(), message.getRequestType()));
+//					}
+//				default:
+//					break;
+//			}
+//		}
+//		if (unis.isEmpty()) {
+//			return Uni.createFrom().voidItem();
+//		}
+//		return Uni.combine().all().unis(unis).discardItems();
+//	}
+
+	public Uni<Void> checkSubscriptions(BatchRequest message) {
+		Collection<SubscriptionRequest> potentialSubs = tenant2subscriptionId2Subscription.row(message.getTenant())
+				.values();
+		List<Uni<Void>> unis = Lists.newArrayList();
+		logger.info("checking subscriptions");
+		for (SubscriptionRequest potentialSub : potentialSubs) {
+			switch (message.getRequestType()) {
+//				case AppConstants.UPDATE_REQUEST:
+//					if (shouldFire(message.getPayload().keySet(), potentialSub)) {
+//						unis.add(localEntityService.getEntityById(message.getTenant(), message.getId(), true).onItem()
+//								.transformToUni(entity -> {
+//									return sendNotification(potentialSub, entity, message.getRequestType());
+//								}));
+//					}
+//					break;
+			case AppConstants.UPSERT_REQUEST:
+			case AppConstants.CREATE_REQUEST:
+				Map<String, Object> queryBody = new HashMap<>();
+				List<Map<String, Object>> entities = new ArrayList<>();
+				queryBody.put("type", "Query");
+				queryBody.put("entities", entities);
+				for (String id : message.getEntityIds()) {
+					Map<String, Object> idMap = new HashMap<>();
+					idMap.put("id", id);
+					entities.add(idMap);
+				}
+				unis.add(localEntityService.getAllByIds(message.getTenant(), queryBody, true, true).onItem()
+						.transformToUni(entityList -> {
+							return sendNotification(potentialSub, entityList, message.getRequestType());
+						}));
+				break;
```

Comment:
Add other cases also as Delete and update

Reply:
done